tests/integration/test_charm.py

```diff
@@ -99,3 +117,57 @@ async def test_application_config(ops_test: OpsTest, get_unit_ip_list) -> None:
             response = requests.get(f"http://{unit_ip}:8080/hello-world", timeout=5)
             assert response.status_code == 200
             assert "Bonjour" in response.text
+
+
+async def test_jvm_config(ops_test: OpsTest, get_unit_ip_list) -> None:
+    """
+    arrange: deploy Spring Boot applications.
+    act: Update the jvm-config to configure JVM heap memory settings.
+    assert: Spring Boot example applications should run with corresponding JVM parameters.
+    """
+    assert ops_test.model
+    jvm_config = "-Xms256m -Xmx512M"
+    await asyncio.gather(
+        ops_test.model.applications[APP_NAME].set_config({"jvm-config": jvm_config}),
+        ops_test.model.applications[BUILDPACK_APP_NAME].set_config({"jvm-config": jvm_config}),
+        ops_test.model.wait_for_idle(apps=[APP_NAME, BUILDPACK_APP_NAME], status="active"),
+    )
+    for name in [APP_NAME, BUILDPACK_APP_NAME]:
+        unit_ips = await get_unit_ip_list(name)
+        for unit_ip in unit_ips:
+            response = requests.get(f"http://{unit_ip}:8080/jvm-arguments", timeout=5)
+            assert response.status_code == 200
+            assert response.json() == jvm_config.split()
+
+
+async def test_invalid_jvm_config(ops_test: OpsTest) -> None:
+    """
+    arrange: deploy Spring Boot applications.
+    act: Update the jvm-config with some invalid values.
+    assert: Spring Boot example applications should enter blocked status.
```

Comment:
how about `Spring Boot example applications enters blocked status.`?

Reply:
Isn't `should` required in the assert statement?

Reply:
I suppose I was saying that `should` is implied by `assert`, up to you